samples/nrf9160/modem_shell/src/gnss/gnss_shell.c

```diff
@@ -522,7 +524,33 @@ static int cmd_gnss_agps(const struct shell *shell, size_t argc, char **argv)
 
 static int cmd_gnss_agps_inject(const struct shell *shell, size_t argc, char **argv)
 {
-	return gnss_inject_agps_data();
+	size_t bin_array_length = 0;
+	char buf[AGPS_CMD_LINE_INJECT_MAX_LENGTH];
```

Comment:
This uses 3.5 kB of stack. While the current stack size is big enough (I guess), I would consider allocating this buffer from heap when needed.

Reply:
Makes sense.